src/map/map.c

```diff
@@ -2924,11 +2939,11 @@ static int map_random_dir(struct block_list *bl, int16 *x, int16 *y)
 	if (dist < 1) dist =1;
 
 	do {
-		int j = 1 + 2*(rnd()%4); //Pick a random diagonal direction
+		enum unit_dir dir = unit_get_rnd_diagonal_dir();
```

Comment:
`uint8` instead of `enum unit_dir`.

Reply:
nope. this is a direction so the enum unit_dir has been added for that purpose.